File: 1010-pairs-of-songs-with-total-durations-divisible-by-60/1010-pairs-of-songs-with-total-durations-divisible-by-60.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        int remainder[] = new int[60];
        int count = 0;
        
        for(int t: time) {
            if(t % 60 == 0) {
                count += remainder[0];
            } else {
                count += remainder[60 - t % 60];
            }
            remainder[t % 60]++;
        }
        return count;
    }
}

File: 1010-pairs-of-songs-with-total-durations-divisible-by-60/1010-pairs-of-songs-with-total-durations-divisible-by-60.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        modCount = collections.Counter(duration % 60 for duration in time)
        return (math.comb(modCount[0], 2) + math.comb(modCount[30], 2) +
                sum(modCount[i] * modCount[60 - i] for i in range(1, 30)))